src/Orleans/Statistics/ThreadTrackingStatistic.cs

```diff
@@ -163,7 +162,7 @@ public void OnStartProcessing()
                 firstStart = false;
                 if (StatisticsCollector.CollectContextSwitchesStats)
                 {
-                    TrackContextSwitches();
+                    logger.IncrementMetric(metricName);        
```

Comment:
I second your coment what will happen if I implement another telemetry consumer ? And I will get just a single call to Context Switches/sec, what is it means for me ? 

Obviosly telemetry data coming into interface should make sense to all consumers and not to be custom fit for one specific implementation.

It just one specific example of a more general leaking abstraction in design of IMetricTelemetryConsumer.  Orleans has many metrics constructed from other metrics but IMetricTelemetryConsumer don't have any general interface to implement this.